### ISSUE TITLE
Replace azure-cosmos by azure-mgmt-cosmosdb

### ISSUE THREAD

**Issue by feluelle** [apache/airflow#9742]
**Description**

This should be a change which not effects the usage of the related provider modules i.e. **refactoring**.

Microsoft actively develops python packages for Azure in https://github.com/Azure/azure-sdk-for-python 
> This repository is for active development of the Azure SDK for Python

Those packages can be recognized by prefix `azure-mgmt-`.

**Comment by morrme** [comment]
Can share more about which tasks are required to complete this request?

**Comment by feluelle** [comment]
Thank you for your interest @morrme I will try to expand the description and what to do to solve this issue. 😊

**Comment by feluelle** [comment]
Hey @morrme sorry I want to get https://github.com/apache/airflow/pull/9747 in before we can start with this, because I would like to reuse the authentication mechanism there.

I am gonna add more information about what we need to change to solve this ticket. :)

I would probably then simply start with replacing the dependency and then fix the `AzureCosmosDBHook` which is the interface for all operators that want to communicate to cosmos db. Since we have unittests for this hook we can change all calls from the previous api/dependency to the new one.

Of course we should system test this changed `AzureCosmosDBHook` as well.

So if I understand you correctly, you want to solve this ticket/issue? - then I am assigning it to you if that's okay?! :)
Do you also have an account to system test it?

**Comment by feluelle** [comment]
I would be happy to support you during this. If you want to reach out to me I am on slack, too (feluelle).

**Comment by morrme** [comment]
@feluelle I appreciate it, but I do not have an account, so I don't think I'm able to help. I saw that there were a few of these but not sure what was required to assist.

**Comment by feluelle** [comment]
@morrme no problem. We can find something different you can work on. If you want we can chat via slack to find something you would like to work on i.e. improve, fix or a feature you would like to see in airflow. WDYT? :)

**Comment by mvfc** [comment]
same as the mgmt-batch one

mgmt libraries are for managing the resources, client libraries (not azure-mgmt- prefixed) should be the ones used for apps like airlfow since we want to interact with e.g. cosmos and not create or modify cosmos resources

**Comment by krishnakaanchan-png** [comment]
This one is already done, only not in the way the description asks.

The provider depends on both packages today, `azure-cosmos>=4.6.0` and `azure-mgmt-cosmosdb>=3.0.0`, and each is used for what it is meant for. `CosmosDBManagementClient` is used in `AzureCosmosDBHook.get_conn` to fetch the account keys through `database_accounts.list_keys`, which is a control plane call. The actual document work, `create_item`, `create_container`, `get_database_client` and the rest, goes through `CosmosClient` which is the data plane SDK.

So replacing `azure-cosmos` with `azure-mgmt-cosmosdb` is not possible, the management SDK has no document APIs. Same point as #9741.

Suggest closing this one also.
